columnflow/production/cms/dy.py

```diff
@@ -0,0 +1,459 @@
+
+        cfg.x.external_files = DotDict.wrap({
+            "dy_weight_sf": "/afs/cern.ch/work/m/mrieger/public/mirrors/external_files/DY_pTll_weights_v1.json.gz",  # noqa
+        })
+
+    *get_dy_weight_file* can be adapted in a subclass in case it is stored differently in the external files.
+
+    The campaign era and name of the correction set (see link above) should be given as an auxiliary entry in the config:
+
+    .. code-block:: python
+
+        cfg.x.dy_weight_config = DrellYanConfig(
+            era="2022preEE_NLO",
+            order="NLO",
+            correction="DY_pTll_reweighting",
+            unc_correction="DY_pTll_reweighting_N_uncertainty",
+        )
+
+    *get_dy_weight_config* can be adapted in a subclass in case it is stored differently in the config.
+    """
+
+    # map the input variable names from the corrector to our columns
+    variable_map = {
+        "era": self.dy_config.era,
+        "order": self.dy_config.order,
+        "ptll": events.gen_dilepton_pt,
+    }
+
+    # initializing the list of weight variations
+    weights_list = [("dy_weight", "nom")]
+
+    # appending the respective number of uncertainties to the weight list
+    for i in range(self.n_unc):
+        for shift in ("up", "down"):
+            tmp_tuple = (f"dy_weight{i + 1}_{shift}", f"{shift}{i + 1}")
+            weights_list.append(tmp_tuple)
+
+    # preparing the input variables for the corrector
+    for column_name, syst in weights_list:
+        variable_map_syst = {**variable_map, "syst": syst}
+
+        # evaluating dy weights given a certain era, ptll array and sytematic shift
+        inputs = [variable_map_syst[inp.name] for inp in self.dy_corrector.inputs]
+        dy_weight = self.dy_corrector.evaluate(*inputs)
+
+        # save the weights in a new column
+        events = set_ak_column(events, column_name, dy_weight, value_type=np.float32)
+
+    return events
+
+
+@dy_weights.init
+def dy_weights_init(self: Producer) -> None:
+    # the number of weights in partial run 3 is always 10
+    if self.config_inst.campaign.x.year not in (2022, 2023):
+        raise NotImplementedError(
+            f"campaign year {self.config_inst.campaign.x.year} is not yet supported by {self.cls_name}",
+        )
+    self.n_unc = 10
+
+    # register dynamically produced weight columns
+    for i in range(self.n_unc):
+        self.produces.add(f"dy_weight{i + 1}_{{up,down}}")
+
+
+@dy_weights.requires
+def dy_weights_requires(self: Producer, reqs: dict) -> None:
+    """
+    Adds the requirements needed the underlying task to derive the Drell-Yan weights into *reqs*.
+    """
+    if "external_files" in reqs:
+        return
+
+    from columnflow.tasks.external import BundleExternalFiles
+
+    reqs["external_files"] = BundleExternalFiles.req(self.task)
+
+
+@dy_weights.setup
+def dy_weights_setup(
+    self: Producer,
+    reqs: dict,
+    inputs: dict,
+    reader_targets: InsertableDict,
+) -> None:
```

Comment:
Just adding the task argument here.

```suggestion
def dy_weights_setup(
    self: Producer,
    task: law.Task,
    reqs: dict,
    inputs: dict,
    reader_targets: law.util.InsertableDict,
) -> None:
```